File: 1248. Count Number of Nice Subarrays/1248. Count Number of Nice Subarrays.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var numberOfSubarrays = function(nums, k) {
    let arraySize = nums.length
    for (let index= 0;index<arraySize;index++){
        if(nums[index]%2==0){
            nums[index]=0
        }else{
            nums[index]=1
        }
    }

    let keyValuePair = {
        0:1
    }
    
    let sum = 0
    let count = 0
    for (let num of nums){
        sum+=num
        let tempDiff = sum-k

        if(keyValuePair[tempDiff]!==undefined){
            count+=keyValuePair[tempDiff]
        }

        if(keyValuePair[sum]!==undefined){
            keyValuePair[sum]+=1
        }else{
            keyValuePair[sum]=1
        }
    }

    return count
};

File: 1248. Count Number of Nice Subarrays/1248. Count Number of Nice Subarrays.py
class Solution:
    def numberOfSubarrays(self, nums: List[int], k: int) -> int:

        curr_count=0
        dic={0:1}
        count = 0
        for num in nums:

            curr_val = 0 if (num%2==0) else 1
            curr_count+=curr_val
            diff = curr_count-k
            count+=dic.get(diff,0)
            dic[curr_count]=dic.get(curr_count,0)+1
            

        return count
